collectors/proc.plugin/sys_fs_btrfs.c

```diff
@@ -736,6 +997,147 @@ int do_sys_fs_btrfs(int update_every, usec_t dt) {
+                                                        netdata_zero_metrics_enabled == CONFIG_BOOLEAN_YES))) {
+            do_commit_stats = CONFIG_BOOLEAN_YES;
+
+            if(unlikely(!node->st_commits)) {
+                char id[RRD_ID_LENGTH_MAX + 1], name[RRD_ID_LENGTH_MAX + 1], title[200 + 1];
+
+                snprintf(id, RRD_ID_LENGTH_MAX, "commits_%s", node->id);
+                snprintf(name, RRD_ID_LENGTH_MAX, "commits_%s", node->label);
+                snprintf(title, 200, "BTRFS Commits");
+
+                netdata_fix_chart_id(id);
+                netdata_fix_chart_name(name);
+
+                node->st_commits = rrdset_create_localhost(
+                        "btrfs"
+                        , id
+                        , name
+                        , node->label
+                        , "btrfs.commits"
+                        , title
+                        , "commits"
+                        , PLUGIN_PROC_NAME
+                        , PLUGIN_PROC_MODULE_BTRFS_NAME
+                        , NETDATA_CHART_PRIO_BTRFS_COMMITS
+                        , update_every
+                        , RRDSET_TYPE_LINE
+                );
+
+                node->rd_commits = rrddim_add(node->st_commits, "commits", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+
+                add_labels_to_btrfs(node, node->st_commits);
+            }
+
+            rrddim_set_by_pointer(node->st_commits, node->rd_commits, node->commits);
+            rrdset_done(node->st_commits);
+
+
+            if(unlikely(!node->st_commit_timings)) {
+                char id[RRD_ID_LENGTH_MAX + 1], name[RRD_ID_LENGTH_MAX + 1], title[200 + 1];
+
+                snprintf(id, RRD_ID_LENGTH_MAX, "commit_timings_%s", node->id);
+                snprintf(name, RRD_ID_LENGTH_MAX, "commit_timings_%s", node->label);
+                snprintf(title, 200, "BTRFS Commit Timings");
+
+                netdata_fix_chart_id(id);
+                netdata_fix_chart_name(name);
+
+                node->st_commit_timings = rrdset_create_localhost(
+                        "btrfs"
+                        , id
+                        , name
+                        , node->label
+                        , "btrfs.commit_timings"
+                        , title
+                        , "ms"
+                        , PLUGIN_PROC_NAME
+                        , PLUGIN_PROC_MODULE_BTRFS_NAME
+                        , NETDATA_CHART_PRIO_BTRFS_COMMIT_TIMINGS
+                        , update_every
+                        , RRDSET_TYPE_LINE
+                );
+
+                node->rd_commit_timings_last = rrddim_add(node->st_commit_timings, "last commit", NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                node->rd_commit_timings_max = rrddim_add(node->st_commit_timings, "max commit", NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                node->rd_commit_timings_total = rrddim_add(node->st_commit_timings, "current commit", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
```

Comment:
Please split this into two charts: don't mix absolute and incremental dimensions.

Reply:
Done.

Reply:
Why? Looks like 2 different metrics to me:
 - (last, max) commit time
 - commit**s** time.

Reply:
Hello @Dim-P I agree with you that we should collect anyway, the unique concern here is to group different kind of metrics in the same chart.

Reply:
Hey @thiagoftsm , we did discuss this with @ilyam8 and @Ferroin and decided to create a new chart that shows the commits time share as a percentage value. See latest commit and let me know what you think.